lib/action_view/component/content_areas.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module ActionView
+  module Component # :nodoc:
+    module ContentAreas
+      extend ActiveSupport::Concern
+
+      included do
+        class_attribute :content_areas, default: []
+        self.content_areas = [] # default doesn't work until Rails 5.2
+      end
+
+      class_methods do
```

Comment:
What do you think about putting this in `Base` instead? This feels like a pretty core part of the library.

Reply:
Yeah we could certainly do that.